src/Microsoft.Framework.PackageManager/Restore/NuGet/PackageUtilities.cs

```diff
@@ -103,5 +103,33 @@ internal static ZipArchiveEntry GetEntryOrdinalIgnoreCase(this ZipArchive archiv
                 }
             }
         }
+
+        internal static async Task<bool> IsValidNupkgAsync(string packageId, string path)
+        {
+            // Acquire the lock on a file before we open it to prevent this process from opening
+            // a file deleted by HttpSource.GetAsync()/HttpSource.InvalidateCacheFile() in another process
+            return await ConcurrencyUtilities.ExecuteWithFileLocked(path, _ =>
+            {
+                if (!File.Exists(path))
+                {
+                    return Task.FromResult(false);
+                }
+
+                using (var stream = File.OpenRead(path))
+                {
+                    try {
+                        using (var archive = new ZipArchive(stream, ZipArchiveMode.Read, leaveOpen: false))
```

Comment:
I think you can just do `new ZipArchive(File.OpenRad(path), ZipArchive..., leaveOpen:false)`. Because you are using `leaveOpen: false` the file stream you are passing should be closed for you.